src/order/basic.lean

```diff
@@ -876,6 +876,16 @@ or_iff_not_imp_left.2 $ λ h,
   ⟨λ a ha₁, le_of_not_gt $ λ ha₂, h ⟨a, ha₁, ha₂⟩,
     λ a ha₂, le_of_not_gt $ λ ha₁, h ⟨a, ha₁, ha₂⟩⟩
 
+/-- If a linear order has no elements `x < y < z`, then it has at most two elements. -/
+lemma eq_or_eq_or_eq_of_forall_not_lt_lt {α : Type*} [linear_order α]
+  (h : ∀ {x y z : α}, x < y → y < z → false) (x y z : α) : x = y ∨ y = z ∨ x = z :=
```

Comment:
I think this `h` is `covby x z`? cc @YaelDillies

Reply:
Ah nevermind, I guess that's not helpful here

Reply:
Yeah, especially because `cover` would need `x < z` too.

Reply:
I can formulate this as `x < y → covby x y` but it'll make it harder to use this lemma.